com.unity.shadergraph/Editor/Data/Graphs/SamplerStateShaderProperty.cs

```diff
@@ -16,24 +16,24 @@ public SamplerStateShaderProperty()
 
         public override PropertyType propertyType => PropertyType.SamplerState;
 
+        // Sampler States cannot be exposed on a Material
         internal override bool isExposable => false;
-        internal override bool isRenamable => false;
 
-        public override TextureSamplerState value
-        {
-            get => base.value;
-            set
-            {
-                overrideReferenceName = $"SamplerState_{value.filter}_{value.wrap}";
-                base.value = value;
-            }
-        }
+        // subgraph Sampler States can be renamed
+        // just the actual properties they create will always have fixed names
+        internal override bool isRenamable => true;
+
+        internal override bool isReferenceRenamable => false;
+
+        // this is the fixed naming scheme for actual samplerstates properties
+        string propertyReferenceName => value.defaultPropertyName;
+        public override string referenceNameForEditing => propertyReferenceName;
 
         internal override bool AllowHLSLDeclaration(HLSLDeclaration decl) => false; // disable UI, nothing to choose
 
         internal override void ForeachHLSLProperty(Action<HLSLProperty> action)
         {
-            action(new HLSLProperty(HLSLType._SamplerState, referenceName, HLSLDeclaration.Global));
+            action(new HLSLProperty(HLSLType._SamplerState, propertyReferenceName, HLSLDeclaration.Global));
```

Comment:
basically any code referring to the actual property should use the propertyReferenceName.